spring-cloud-data-rest-client/src/main/java/org/springframework/cloud/data/rest/client/CounterTemplate.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2015 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.rest.client;
+
+import org.springframework.cloud.data.rest.resource.CounterResource;
+import org.springframework.cloud.data.rest.resource.MetricResource;
+import org.springframework.hateoas.PagedResources;
+import org.springframework.hateoas.ResourceSupport;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Implementation for {@link CounterOperations} that interacts with the Spring Cloud Data REST API.
+ *
+ * @author Eric Bottard
+ */
+public class CounterTemplate implements CounterOperations {
+
+	public static final String COUNTERS_COLLECTION_RELATION = "counters";
```

Comment:
Nope, can be private (but will need to be shared with the other template if we follow your idea)